### PR TITLE
Re-do docker image

### PR DESCRIPTION
**Pull request by romcheg**
Build a production-ready image by installing ralph-core
package to vanila ubuntu:bionic base image.

Co-authored-by: Maciej Skorczewski <maciej.skorczewski@allegro.pl>
Co-authored-by: Roman Prykhodchenko <me@romcheg.me>

### CODE REVIEW

File: docker/Dockerfile-prod [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ARG RALPH_VERSION=""
+ENV PATH=/opt/ralph/ralph-core/bin/:$PATH
+ENV RALPH_CONF_DIR="/etc/ralph"
+ENV RALPH_LOCAL_DIR="$RALPH_LOCAL_DIR"
+ENV RALPH_IMAGE_TMP_DIR="/tmp"
+
+# Default superuser paramethers
+ENV RALPH_ADMIN_USERNAME=ralph
+ENV RALPH_ADMIN_PASSWORD=ralph
+ENV RALPH_ADMIN_EMAIL=admin@ralph.com
```
Comment (szymi-) [inline docker/Dockerfile-prod]: Email is not needed. Who is or will be the owner of ralph.com domain anyway?

File: docker/provision/init-ralph.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#! /bin/bash
+set -eu
+
+
+COMMAND="$1"
+VALID_COMMANDS=(init upgrade)
+
+if [[ " ${VALID_COMMANDS[*]} " != *"$COMMAND"* ]]; then
```
Comment (szymi-) [inline docker/provision/init-ralph.sh]: How about

```suggestion
isValidCommand () {
  VALID_COMMANDS=(init upgrade)
  match="$1"
  for i in ${VALID_COMMANDS[@]}; do [[ "$i" == "$match" ]] && return 0 ; done
  return 1
}

if ! isValidCommand $i ; then exit 1 ; fi
```

File: docker/Dockerfile-prod (hunk 1 above)
Comment (romcheg) [reply docker/Dockerfile-prod]: I have removed these values from the dockerfile and introduces them as default values in the script. This way users will be able to set up default user on initialization.

**Review by vi4m — DISMISSED** [review]
Well, it looks ok, but it needs at least second review, since it's big TBH

**Review by szymi- — DISMISSED** [review]

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  web:
+    image: allegro/ralph:latest
+    restart: always
+    ports:
+      - "8000"
+    volumes:
+      - ralph_media:/var/local/ralph/media
+      - ralph_static:/usr/share/ralph/static
+    links:
+      - db:db
```
Comment (ar4s) [inline docker/docker-compose.yml]: It's equivalent to: 
```
- db
- redis
- nginx
```

File: docker/provision/init-ralph.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    for i in ${VALID_COMMANDS[@]}; do
+        [[ "$i" == "$match" ]] && return 0
+    done
+
+    return 1
+}
+
+
+if ! is_valid_command "$COMMAND" ; then
+    echo "Usage: ${0} {init|upgrade}"
```
Comment (ar4s) [inline docker/provision/init-ralph.sh:21]: Where is ``upgrade`` command?

File: docker/provision/start-ralph.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+/opt/ralph/ralph-core/bin/gunicorn --daemon -c "${RALPH_LOCAL_DIR}/gunicorn.ini" ralph.wsgi
+
+while [ ! -f /var/log/ralph/ralph.log ] || [ ! -f /var/log/ralph/gunicorn.error.log ]
```
Comment (ar4s) [inline docker/provision/start-ralph.sh]: Can we use ``wait-for-it.sh`` for that?

File: docker/ralph.conf.nginx [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# Example NGINX config for Ralph working with Dockerized Ralph out of the box
+
+server {
+    listen 80;
+    server_name ralph;
+
+    location /static {
+       alias /opt/static;
```
Comment (ar4s) [inline docker/ralph.conf.nginx]: add
```
expires 1y; # unmutable files
add_header Pragma public;
add_header Cache-Control "public";
access_log off; # optional
```

**Review by mkurek — DISMISSED** [review]

File: docker/ralph.conf.nginx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    location /media {
+       alias /opt/media;
+       add_header Content-disposition "attachment";
```
Comment (ar4s) [inline docker/ralph.conf.nginx]: ```access_log off;```

File: docker/docker-compose.yml (hunk 3 above)
Comment (romcheg) [reply docker/docker-compose.yml]: Done

File: docker/provision/init-ralph.sh (hunk 4 above)
Comment (romcheg) [reply docker/provision/init-ralph.sh:21]: It requires no additional actions atm.

File: docker/provision/start-ralph.sh (hunk 5 above)
Comment (romcheg) [reply docker/provision/start-ralph.sh]: Good question. Sounds like something we should do. Let me test that.

File: docker/ralph.conf.nginx (hunk 6 above)
Comment (romcheg) [reply docker/ralph.conf.nginx]: Done

File: docker/ralph.conf.nginx (hunk 7 above)
Comment (romcheg) [reply docker/ralph.conf.nginx]: Done

**Review by Blejwi — DISMISSED** [review]

File: docker/provision/start-ralph.sh (hunk 5 above)
Comment (romcheg) [reply docker/provision/start-ralph.sh]: Seems to work

**Review by ar4s — APPROVED** [review]

**Review by mkurek — APPROVED** [review]

**Pull request merged**